src/Lorentz/Contracts/RegistryDAO/Types.hs

```diff
@@ -0,0 +1,183 @@
+  , reLastUpdated :: Timestamp
+  } deriving stock (Eq, Generic)
+  deriving anyclass (IsoValue)
+
+instance (HasAnnotation k, HasAnnotation v)
+  => HasAnnotation (RegistryEntry k v) where
+    annOptions = DAO.baseDaoAnnOptions
+
+instance (NiceComparable k, Ord k, IsoValue v, TypeHasDoc k, TypeHasDoc v)
+  => TypeHasDoc (RegistryEntry k v) where
+    typeDocMdDescription =
+      "Describe the value in registry map. It represents the actual item of the registry as `Maybe v`. \
+      \`None` represents the deletion of item and `Some v` represents the existence of an item.\
+      \It also contains the last proposal's agora post id that affects this item and its last updated time."
+    typeDocMdReference = poly2TypeDocMdReference
+    typeDocHaskellRep = concreteTypeDocHaskellRep @(RegistryEntry ByteString ByteString)
+    typeDocMichelsonRep = concreteTypeDocMichelsonRep @(RegistryEntry ByteString ByteString)
+
+
+data RegistryDaoProposalMetadata k v
+  = NormalProposalType (NormalProposal k v)
+  | ConfigProposalType ConfigProposal
+  deriving stock (Generic)
+
+type IsoRegistryDaoProposalMetadata k v = (NiceComparable k, Ord k, IsoValue k, KnownValue v)
+
+instance IsoRegistryDaoProposalMetadata k v
+  => (IsoValue (RegistryDaoProposalMetadata k v))
+
+instance (HasAnnotation k, HasAnnotation v)
+  => HasAnnotation (RegistryDaoProposalMetadata k v) where
+    annOptions = DAO.baseDaoAnnOptions
+
+instance (TypeHasDoc k, TypeHasDoc v)
+  => TypeHasDoc (RegistryDaoProposalMetadata k v) where
+    typeDocMdDescription =
+      "Describe the metadata of a proposal in Registry DAO. In Registry DAO, there are 2 types of \
+      \proposals: a registry proposal, represented as `NormalProposal k v` and a configuration proposal \
+      \represented as `ConfigProposal`."
+    typeDocMdReference = poly2TypeDocMdReference
+    typeDocHaskellRep = concreteTypeDocHaskellRep @(RegistryDaoProposalMetadata ByteString ByteString)
+    typeDocMichelsonRep = concreteTypeDocMichelsonRep @(RegistryDaoProposalMetadata ByteString ByteString)
+
+
+-- | A registry proposal. It will update registry list in contract storage when got accepted.
+data NormalProposal k v = NormalProposal
+  { npAgoraPostId :: AgoraPostId
+  , npDiff :: [RegistryUpdate k v]
+  }
+  deriving stock (Generic)
+
+instance IsoRegistryDaoProposalMetadata k v
+  => (IsoValue (NormalProposal k v))
+
+instance (HasAnnotation k, HasAnnotation v)
+  => HasAnnotation (NormalProposal k v) where
+    annOptions = DAO.baseDaoAnnOptions
+
+instance (TypeHasDoc k, TypeHasDoc v)
+  => TypeHasDoc (NormalProposal k v) where
+    typeDocMdDescription =
+      "Describe a proposal for add/update/delete an item in the registry."
+    typeDocMdReference = poly2TypeDocMdReference
+    typeDocHaskellRep = concreteTypeDocHaskellRep @(NormalProposal ByteString ByteString)
+    typeDocMichelsonRep = concreteTypeDocMichelsonRep @(NormalProposal ByteString ByteString)
```

Comment:
Since you mention `ByteString`s as examples, you also need to include them into dependencies, otherwise there will be a reference to a description of this type but the desription itself won't be included. The same for `RegistryUpdate` type.

You can check how it is done for View: https://gitlab.com/morley-framework/morley/blob/master/code/lorentz/src/Lorentz/Macro.hs#L661